packages/pubsub/src/publisher.js

```diff
@@ -0,0 +1,179 @@
+/*!
+ * Copyright 2014 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*!
+ * @module pubsub/publisher
+ */
+
+'use strict';
+
+var arrify = require('arrify');
+var common = require('@google-cloud/common');
+var extend = require('extend');
+var is = require('is');
+
+/**
+ *
+ */
+function Publisher(topic, options) {
+  this.topic = topic;
+  this.api = topic.api;
+
+  this.inventory = {
+    queued: [],
+    inFlight: []
+  };
+
+  Object.defineProperty(this.inventory, 'queueSize', {
+    get: function() {
+      return this.queued.reduce(function(size, message) {
+        return size + message.data.length;
+      }, 0);
+    }
+  });
+
+  this.settings = extend(true, {
```

Comment:
Yes! The boolean tells `extend` to do a deep copy, so any batch settings should be respected.